src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/auth/securityconfigurer/ApiWebSecurityConfigurerAdapter.java

```diff
@@ -64,6 +64,10 @@ protected void configure(HttpSecurity http) throws Exception {
         filter.setExceptionIfHeaderMissing(false); // exception at this point
         // would cause http 500, we want http 401
         http
+            .authorizeRequests()
+                .antMatchers("/api/**/openapi")
+                .permitAll()
+                .and()
```

Comment:
This now has two `authorizeRequests()` blocks/calls. Are you sure this is legal way to use of the API? I am not sure how the two blocks interact with the other parts of the configuration. It seems to work in quick tests, but the configuration seems a bit unorthodox, and I am not sure if it is robust when we or Spring make some modifications?

There's also some parts that are not relevant for the always-public endpoint, like `ApiKeyAuthenticationManager` and `RequestHeaderAuthenticationFilter`, and possibly csr and session related configuration if this API is not meant to be used via frontend.

I think a structure with less question marks could be to introduce a new `WebSecurityConfigurerAdapter` reserved only for unauthenticated API calls. Probably put it in front of `ApiWebSecurityConfigurerAdapter` (index 2, move others accordingly)

Reply:
Unauthenticated endpoints are now configured in the separate method `ApiWebSecurityConfigurerAdapter#configure(WebSecurity web)`